packages/astro/src/assets/runtime.ts

```diff
@@ -0,0 +1,54 @@
+import {
+	createComponent,
+	render,
+	spreadAttributes,
+	unescapeHTML,
+} from '../runtime/server/index.js';
+import type { ImageMetadata } from './types.js';
+import { makeNonEnumerable, normalizeProps } from './utils/svg.js';
+
+export interface SvgComponentProps {
+	meta: ImageMetadata;
+	attributes: Record<string, string>;
+	children: string;
+}
+
+/**
+ * Make sure these IDs are kept on the module-level so they're incremented on a per-page basis
+ */
+let ids = 0;
```

Comment:
IIUC, if this is on the module-level (and also globally as a singleton), wouldn't this value not be incremented on a per-page basis? If per page, I think it should be bound to the `result` from `createComponent` as a single `result` is created when rendering per page. So maybe we can use a WeakMap here to track the ids per `result`, which we can also use it to track the first render instead of using `WeakSet<Response>`?

Reply:
I'm not quite following how to use the `WeakMap` here.

Reply:
The suggestion here is to use `const ids = new WeakMap<SSRResult, number>` instead to track the `id` per `result` (from `createComponent`)

Reply:
I'm having trouble getting this to work. I'll give it a shot again tomorrow. Maybe I'm just tired 😅

Reply:
I'll help you out :)

Reply:
Here [`6ab0f1a` (#12067)](https://github.com/withastro/astro/pull/12067/commits/6ab0f1acc2b0792845ff9e75f0bca022f1593a84)

Reply:
We don't need the `counter` variable there. What I mean is that the WeakMap values are the counter itself that starts from 0 on first assigned.

```js
if (!ids.has(result)) ids.set(result, 0)
const id = ids.get(result)
ids.set(result, id + 1)
```

Reply:
I'll give it a shot but I was having issues previously with this.